syntax = "proto3";
package esto_rpc;

service Esto {
    rpc StoreRecord (StoreRequest) returns (StoreReply);
    rpc ReadRecord (ReadRequest) returns (ReadReply);
}

message StoreRequest {
    string entity_id = 1 ;
    string entity_type = 2;

    string event_name = 3;
    string event_data = 4;
}

message StoreReply {
    string message = 1;
}

message ReadRequest {
    string entity_id = 1 ;
}

message ReadReply {
    string entity_data = 1;
}